nupic/research/frameworks/pytorch/imagenet/imagenet_experiment.py

```diff
@@ -392,20 +395,33 @@ def create_lr_scheduler(cls, config, optimizer, total_batches):
                 steps_per_epoch=total_batches)
 
     @classmethod
-    def create_train_dataloader(cls, config):
+    def create_loaders(cls, config):
+        """Create train, val, and test dataloaders."""
+
+        dataset_manager_class = config.get("dataset_manager_class", None)
+        if dataset_manager_class is None:
+            raise ValueError("Must specify 'dataset_manager_class' in config.")
+        dataset_args = config.get("dataset_args", {})
+        dataset_manager = dataset_manager_class(**dataset_args)
+
+        train_set = dataset_manager.get_train_dataset()
+        val_set = dataset_manager.get_val_dataset()
+        test_set = dataset_manager.get_test_dataset()
```

Comment:
Maybe you could refactor this code as a class method as well (`create_dataset_manager`). It would allow other code to create `dataset_manager` accessing the experiment's datasets without instantiating dataloaders loaders. This could be useful for testing and other reporting scripts.

Reply:
Per your comment and Marcus's, there's now a create_datasets_func that you pass to the config.